### PR TITLE
Issue #2566871 by Primsi: Better handle cardinality

### PR DESCRIPTION
**Pull request by primsi**

### CODE REVIEW

File: js/entity_browser.common.js [hunk 1]
```diff
@@ -20,11 +20,46 @@
...
+        else {
+          existing_entities_array.splice(0, added_entities_array.length);
+          existing_entities_array = _.union(existing_entities_array, added_entities_array);
+        }
+      }
+
+      entity_ids = existing_entities_array.join(' ');
+    }
+    else {
+      entity_ids = entity_ids + added_entities_array.join(' ');
```
Comment (slashrsm) [inline js/entity_browser.common.js]: Could this be

```
entity_ids = added_entities_array.join(' ');
```

File: js/entity_browser.common.js [hunk 2]
```diff
@@ -20,11 +20,46 @@
...
+    if (cardinality != -1 && added_entities_array.length > cardinality) {
+      added_entities_array.splice(0, added_entities_array.length - cardinality);
+    }
+
     // Update value form element with new entity IDs.
-    var entity_ids = $(this).parent().parent().find('input[type*=hidden]').val().split(" ");
-    var selected_entity_ids = $.map(entities, function(item) {return item[0]});
-    var entity_ids = _.union(entity_ids, selected_entity_ids);
-    $(this).parent().parent().find('input[type*=hidden]').val(entity_ids.join(" "));
+    var entity_ids = $(this).parent().parent().find('input[type*=hidden]').val();
```
Comment (neardark) [inline js/entity_browser.common.js:38]: You might look into using parents() or closest() here. See https://api.jquery.com/closest/ or https://api.jquery.com/parents/

File: js/entity_browser.common.js [hunk 3]
```diff
@@ -20,11 +20,46 @@
...
+        }
+      }
+
+      entity_ids = existing_entities_array.join(' ');
+    }
+    else {
+      entity_ids = entity_ids + added_entities_array.join(' ');
+    }
+
+    $(this).parent().parent().find('input[type*=hidden]').val(entity_ids);
```
Comment (neardark) [inline js/entity_browser.common.js:62]: Maybe parents() or closest() here?

File: js/entity_browser.common.js (hunk 2 above)
Comment (primsi) [inline js/entity_browser.common.js:38]: This is existing code. Should I create a separate ticket for that? Same down below.

File: src/Plugin/EntityBrowser/Display/IFrame.php [hunk 4]
```diff
@@ -83,7 +90,7 @@ class IFrame extends DisplayBase implements DisplayRouterInterface {
   public function __construct(array $configuration, $plugin_id, $plugin_definition, EventDispatcherInterface $event_dispatcher, RouteMatchInterface $current_route_match, UuidInterface $uuid, Request $request, CurrentPathStack $current_path) {
     parent::__construct($configuration, $plugin_id, $plugin_definition, $event_dispatcher);
     $this->currentRouteMatch = $current_route_match;
-    $this->uuid = $uuid;
+    $this->uuidService = $uuid;
```
Comment (slashrsm) [inline src/Plugin/EntityBrowser/Display/IFrame.php]: Just one nit-pik. "$uuidGenerator" seems better name. What do you think?

**Pull request merged**